index.js

```diff
@@ -139,46 +190,70 @@ const line = []
 
 // ***** Scope & Closures - Question 1 *****
 
+let ticketNumber = 0 
+
+function takeATicketNumber(line) {
+  ticketNumber += 1 
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
+// *** HAD TO GET A LITTLE HELP FROM SOLUTION TAB FOR THIS ONE 
+// *** Forgot --> let ticketNumber = 0 
+
+
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+
+function ticketNumberGeneratorFunc() {
+  let number = 0 
+
+  return 'Welcome you are ticket number ${number}'
+}
+
+// *** Couldn't get this one! A little confused on this topic still!! 
+
+
```

Comment:
This one's a tricky bonus one! Check this Scrimba out [https://scrimba.com/scrim/cLgEmZUZ](https://scrimba.com/scrim/cLgEmZUZ) if you haven't already if you want to try it again. The key here is to learn about higher order functions and closures. Think of the `ticketNumberGeneratorFunc` as a function factory. It should return a function that is in charge of increasing the ticket number and pushing the new ticket number into the array.

Reply:
Sounds good! I will definitely check it out and revise. Appreciate all of your comments and feedback!